test_suite/src/tester.rs

```diff
@@ -109,41 +369,41 @@ pub fn test_regex(regex_input: &DbEntry, code: &mut Code) -> anyhow::Result<Test
             Vec::new()
         }
     };
-
-    let (random_succ_pass, random_unsucc_pass) =
+    let (random_samples_correct, random_samples_wrong) =
         evaluate_test_set(code, &regex_input.regex.complete_regex(), &random_samples)?;
-    let (passes_that_passed, passes_that_not_passed) = evaluate_test_set(
-        code,
-        &regex_input.regex.complete_regex(),
-        &regex_input.samples_pass,
-    )?;
-    let (failures_that_failed, failures_that_not_fail) = evaluate_test_set(
-        code,
-        &regex_input.regex.complete_regex(),
-        &regex_input.samples_fail,
-    )?;
+    Ok((random_samples_correct, random_samples_wrong))
+}
 
-    let successful_passes = random_succ_pass
-        .iter()
-        .chain(passes_that_passed.iter())
-        .cloned()
-        .collect();
-    let unsuccessful_passes = random_unsucc_pass
-        .iter()
-        .chain(passes_that_not_passed.iter())
-        .cloned()
-        .collect();
-
-    let test_result = TestResult::new(
-        failures_that_failed,
-        successful_passes,
-        failures_that_not_fail,
-        unsuccessful_passes,
-    );
+fn run_single_standard_test(
```

Comment:
I think that maybe we should remove this function and `ru_single_gen_substr_test` and just call `run_single_test` given that this function is not doing something additional to call another function. Maybe to make clear why are we passing this set of parameters, we can add comments explaining why some parameters are `None` or `Some`

Reply:
that's what I did first and I think this is cleaner and less error prone